File: frontend/src/api/mappers/Project.mapper.ts
import dayjs from 'dayjs';

import { patternMapper } from '@/api/mappers/Pattern.mapper';

import { type IProject } from '@/models/Project.model';
import { type IApiProject } from '@/api/api-models/ApiProject.type';

export const projectMapper = {

    map(project: IApiProject): IProject {
        return {
            pattern: patternMapper.map(project.pattern),
            purchasedAt: dayjs(project.purchasedAt),
            pausePositionX: project.pausePositionX,
            pausePositionY: project.pausePositionY,
        };
    },

};

File: backend/ClickStitch/Models/Mappers/ProjectMapper.cs
﻿using Data.Records;

namespace ClickStitch.Models.Mappers;

public static class ProjectMapper
{
    public static ProjectModel Map(UserPatternRecord userPattern) => new()
    {
        Pattern = PatternMapper.Map(userPattern.Pattern),
        Reference = userPattern.Reference,
        PurchasedAt = userPattern.CreatedAt,
        PausePositionX = userPattern.PausePositionX,
        PausePositionY = userPattern.PausePositionY
    };
}